pkg/volume/storageos/storageos.go

```diff
@@ -0,0 +1,786 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageos
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/mount"
+	kstrings "k8s.io/kubernetes/pkg/util/strings"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/util"
+	"k8s.io/kubernetes/pkg/volume/util/volumehelper"
+)
+
+// ProbeVolumePlugins is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.VolumePlugin {
+	return []volume.VolumePlugin{&storageosPlugin{nil}}
+}
+
+type storageosPlugin struct {
+	host volume.VolumeHost
+}
+
+var _ volume.VolumePlugin = &storageosPlugin{}
+var _ volume.PersistentVolumePlugin = &storageosPlugin{}
+var _ volume.DeletableVolumePlugin = &storageosPlugin{}
+var _ volume.ProvisionableVolumePlugin = &storageosPlugin{}
+
+const (
+	storageosPluginName = "kubernetes.io/storageos"
+	storageosDevicePath = "/var/lib/storageos/volumes"
+	defaultAPIAddress   = "tcp://localhost:5705"
+	defaultAPIUser      = "storageos"
+	defaultAPIPassword  = "storageos"
+	defaultAPIVersion   = "1"
+	defaultFSType       = "ext4"
+	defaultNamespace    = "default"
+	secretLocatorPrefix = "storageos-secretref"
+)
+
+func getPath(uid types.UID, volNamespace string, volName string, pvName string, host volume.VolumeHost) string {
+	if len(volNamespace) != 0 && len(volName) != 0 && strings.Count(volName, ".") == 0 {
```

Comment:
The volNamespace and volName are supplied by the user.  This goes back to my issue with the unmount; from what I can tell the unmount only gets the podUID and pvName, but we need the volName and volNamespace (and API credentials) to reference the volume in our API to remove the mount reference.  Encoding them in the mountpath was a way to get the volNamespace & volName.

Unless there's another way, since we're not able to get the creds I could remove this too.